src/Provider/Yandex/Model/YandexAddress.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Yandex\Model;
+
+use Geocoder\Model\Address;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class YandexAddress extends Address
+{
+    /**
+     * @var string|null
+     */
+    private $precision;
+
+    /**
+     * The name of this location.
+     *
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @return null|string
+     */
+    public function getPrecision()
+    {
+        return $this->precision;
+    }
+
+    /**
+     * @param null|string $precision
+     *
+     * @return YandexAddress
+     */
+    public function setPrecision($precision)
```

Comment:
[Wikipedia](https://en.wikipedia.org/wiki/Builder_pattern) states that the builder pattern solves the the [telescoping constructor anti-pattern](http://www.captaindebug.com/2011/05/telescoping-constructor-antipattern.html#.WSgLeFKB1-0) problem. C#, for example, in the past had no default values for properties and default values for arguments, so there were 2 ways to make default values: constructor overloading and the builder pattern. Regarding to your case with `Address` I don't see why you need the builder.

Technically, it just replaces
```php
$a = ...;
$b = ...;
$c = ... ? ... : null;

$address = new Address($a, $b, $c);
```
with
```php
$builder->setA(...);
$builder->setB(...);
$builder->setC(... ? ... : null);

$address = $builder->build();
```

What problem does the `LocationBuilder` solve?